lib/hsa/kernels/rfi_chime_input_sum.cl

```diff
@@ -55,16 +67,33 @@ rfi_chime_inputsum(
         uint address = gy + gz * gy_size;
         if (n * N == 0){
             output[address] = -1.0;
+            output_var[address] = -1.0;
             output_mask[address] = 1;
         } else {
             float new_sq_power_across_input = sq_power_across_input[0] * cf * cf;
             float SK = ((n + 1) / (n - 1)) * ((new_sq_power_across_input / (n * N)) - 1);
+
+            // Calculate the truncation bias correction for the SK value
+            const float var = var_across_input[0] / (n * N);
+            const float rms = sqrt((double)var);
+            float sk_correction = 0.f;
+            float rms_pow = 1.f;
+            for(int i=0; i<NUM_COEFFS; i++) {
+                sk_correction += bias_coeffs[i] * rms_pow;
+                rms_pow *= rms;
+            }
+
+            // Correct SK for truncation bias
+            SK -= sk_correction * trunc_bias_switch;
```

Comment:
I would do an `if` on the `trunc_bias_switch`  then you can control if we need to do extra work in this kernel or not.  Since all the work units will follow the same `if` path, there is shouldn't be a performance hit doing it that way.